File: code-coaches/medium/name-buddy/NameBuddy.cs
using System;

class NameBuddy
{
    static bool ContainsNameBuddy(string names, string myName)
    {
        string[] namesSplit = names.Split(' ');

        foreach (string name in namesSplit)
        {
            if (name[0] == myName[0])
            {
                return true;  // Found a name buddy
            }
        }

        return false;  // No name buddy found
    }

    static void Main(string[] args)
    {
        string names = Console.ReadLine();
        string myName = Console.ReadLine();

        if (ContainsNameBuddy(names, myName))
            Console.WriteLine("Compare notes");
        else
            Console.WriteLine("No such luck");
    }
}

File: code-coaches/medium/name-buddy/NameBuddy.java
import java.util.Scanner;

public class NameBuddy {

    public static boolean containsNameBuddy(String names, String myName) {
        String[] namesSplit = names.split(" ");

        for (String name : namesSplit) {
            if (name.charAt(0) == myName.charAt(0)) {
                return true;  // Found a name buddy
            }
        }

        return false;  // No name buddy found
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String names = scanner.nextLine();
        String myName = scanner.nextLine();

        if (containsNameBuddy(names, myName))
            System.out.println("Compare notes");
        else
            System.out.println("No such luck");

        scanner.close();
    }

}
